cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -708,9 +709,17 @@ public int createApplicationVersion(ApplicationId id, ApplicationMeta appMeta, b
     }
 
     // Add a new version of the app
-    writeApplication(id.getNamespace(), id.getApplication(), id.getVersion(), appMeta.getSpec(),
-        appMeta.getChange(),
-        appMeta.getSourceControlMeta(), markAsLatest);
+    try {
+      writeApplication(id.getNamespace(), id.getApplication(), id.getVersion(), appMeta.getSpec(),
+          appMeta.getChange(),
+          appMeta.getSourceControlMeta(), markAsLatest);
+    } catch (FieldLimitExceededException e) {
+      LOG.error(
+          "Field {} exceeds spanner cell limit even after compression, size : {} limit : {}",
+          e.getTableId() + "." + e.getField(), e.getSize(), e.getLimit());
+      throw new IllegalArgumentException(
+          "Deployment failed: The pipeline size exceeds the supported limit.");
```

Comment:
nit: use better error message for the exception like "Application deployment failed as size exceeds the supported limit of %d characters"

Use the term 'application' instead of 'pipeline' in these error messages.